### ISSUE TITLE
GRAPHS-2015 and TSP-LION-2015?

### ISSUE THREAD

**Issue by mlindauer** [coseal/aslib_data#16]
@larskotthoff what's the status of GRAPHS-2015 and TSP-LION-2015?
I think that you are the author of both scenarios, right?

Besides updating the description.txt format, is there anything else to do for these scenarios?

Cheers,
Marius

**Comment by mlindauer** [comment]
I updated the description.txt files of both scenarios and fixed some minor issues in GRAPHS-2015.

I also found that the TSP-LION-2015 scenario is actually from @kerschke.
@kerschke Can we merge the TSP-LION-2015 scenario into the master branch?

Regarding the GRAPHS-2015 scenario, the following TODOs are open
* the columns in algorithm_runs.arff are wrongly sorted
* a readme.txt is missing. 
@larskotthoff Could you please fix these two issues?

Cheers,
Marius

**Comment by larskotthoff** [comment]
Thanks, I've done that and added a citation.bib.

**Comment by mlindauer** [comment]
I fixed the algorithm_runs.arff by imputing the missing values with the running time cutoff -- @larskotthoff is this ok for you?
(Furthermore, I fixed again the header of feature_runstatus.arff)

@larskotthoff could you please add in the readme what kind of runtime you measured? I would guess it is not seconds since the running time cutoff would be ~3.2 years ;-)

If this is also fixed, I would say that we can also merge this scenario in the master branch.

**Comment by mlindauer** [comment]
@kerschke and @larskotthoff  I added default feature steps to the description.txt of TSP-LION2015 since they were missing. Please check whether you agree with these default steps.

**Comment by kerschke** [comment]
@larskotthoff what do you think of @mlindauer's suggestion? I am not that sure, which ones would actually be reasonable default feature sets, but I would rather replace the probing features by the ubc.cheap features.. Would you agree?
And since you started the algorithms, could you actually also provide the algorithm metainfo?

**Comment by mlindauer** [comment]
sorry for posting so often today.
I tried to reproduce your vbs and single best scores as you report them in your paper:
https://www.cs.ubc.ca/~larsko/papers/tsplion.pdf

Unfortunately, my numbers do not agree with your numbers. 
Virtual Best Algorithm -- mine: 21.913917 --  your: 18.52
Best Single Algorithm (eax.restart)	-- mine: 117.236988 -- your: 104.01

How have you computed your numbers; in particular, how have you handled the repeated runs?
I simply computed the mean over the repetitions.

**Comment by kerschke** [comment]
We first aggregated the runs, i.e., if more than 5 of the 10 runs were successful, we took the median (= the mean of the 5th and 6th smallest runtime), otherwise the run is failed.
After aggregating those values accordingly, I receive a vbs of 18.52 (on my local data).
In any case, I will double-check the data once more before giving my "ok" and I plan to do this, by the end of this week.

**Comment by mlindauer** [comment]
ah, ok. using median instead of mean, I get in fact the same numbers. thanks!

**Comment by larskotthoff** [comment]
Thanks for checking all of this. I've added the explanation for the runtime measurements.

I agree with @kerschke that the default step should be ubc_cheap -- I've changed this accordingly.

I've added the solver configurations, but some of them need to be set in configuration files and require additional arguments (in <>)...

I remember that the aggregation for the TSP dataset was the reason we didn't merge it immediately. We could have only the aggregated data in there so that the results from the paper are directly reproducible. What do you think?

**Comment by mlindauer** [comment]
> I remember that the aggregation for the TSP dataset was the reason we didn't merge it immediately. We could have only the aggregated data in there so that the results from the paper are directly reproducible. What do you think?

I disagree. We added the column "repetition" in the first place because we wanted to expose the full data to AS practitioners. Hiding them now would be in contrast to spirit of the format.
I would propose that you describe your approach in the readme such that everyone know how to reproduce your results.
(even I was able to reproduce the results after knowing the little details ;-) )

**Comment by mlindauer** [comment]
I moved GRAPHS-2015 into the master branch.

**Comment by larskotthoff** [comment]
Thanks! I'm happy to leave the TSP scenario as is.

**Comment by mlindauer** [comment]
> I'm happy to leave the TSP scenario as is.
I'm waiting only for the approval from @kerschke and then, I will also move this scenario to the master branch.

**Comment by kerschke** [comment]
I've just added a note within the readme, stating how we aggregated the replications of our algorithm runs. And I've also double-checked the numbers within algorithm_runs.arff and cv.arff. To be honest, I did not double-check all the feature-related arff-files, but I would expect that they are correct as well (until someone else stumbles across something, which might indicate that there could be a bug).

Therefore, I'm happy with the TSP scenario as well 👍
